Core/Items/PoTItemHelper.cs

```diff
@@ -130,6 +130,35 @@ private static void RollAffixes(Item item)
 		}
 	}
 
+	/// <summary>
+	///		Adds a new random affix to an item. This is used for things like the ascendant shard.
+	/// </summary>
+	/// <param name="item"></param>
+	public static void AddNewAffix(Item item)
+	{
+		PoTInstanceItemData data = item.GetInstanceData();
+		PoTStaticItemData staticData = item.GetStaticData();
+		if (data.Affixes.Count >= GetAffixCount(item))
+		{
+			return;
+		}
+
+		ItemAffixData chosenAffix = AffixRegistry.GetRandomAffixDataByItemType(data.ItemType);
```

Comment:
I forget which method but there's a method I think in this file that you seem to have copied this code from. It'd be a bit nicer to split this copied code into a method and make both this new method and the original use that instead, assuming they are the same.

Reply:
I didn't copy this in, but can search around and see if there is something like it. I assume the roll probably does this in a loop.

Reply:
Lmfao, that EXACT code was right above it and it was unintentionally. That said fixed up the code above it to just call it